### PR TITLE
Edit mode controller mapping using new platform state variables

### PR DESCRIPTION
**Pull request by dback2**
Add a controller mapping for edit mode keyboard shortcuts using new platform-specific state variables in Application to allow for separating the delete selected entities shortcut to be the actual Delete key on PC (next to End) and the Backspace key on Mac.

Also remove the '[' keyboard shortcut for entering edit camera, and allow 'F' key to enter edit camera without any selected entities to focus on.

Fixes: https://highfidelity.manuscript.com/f/cases/6395

Test Plan:
- Enter Interface in desktop mode on PC and open Create app
- Verify the following keyboard shortcuts work as expected:
Delete key = Delete selected entities
Ctrl + D on PC and Cmd + D on Mac = Deselect selected entities
T = Toggle between world and local space edit handles
F = Enter edit camera or focus on any selected entities
G = Align grid to bottom of selected entities (requires grid set to visible on Grid tab)
- Enter Interface on Mac and open Create app
- Verify the above keyboard shortcuts work as expected

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12432/HighFidelity-Beta-PR13888-f81fe6b.apk

**Comment by danteruiz** [comment]
Can you elaborate on this `separate Delete key between Delete on PC and Backspace on Mac.`

**Comment by dback2** [comment]
^ adjusted description

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12432/HighFidelity-Beta-PR13888-f81fe6b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12432/HighFidelity-Beta-PR13888-f81fe6b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12432/HighFidelity-Beta-Interface-PR13888-f81fe6b.exe

File: scripts/system/edit.js [hunk 1]
```diff
@@ -1893,6 +1874,52 @@ var keyReleaseEvent = function (event) {
...
+function gridKey(value) {
+    if (value === 0) { // on release
+        if (selectionManager.hasSelection()) {
+            grid.moveToSelection();
+        }
+    }
+}
+var mapping = Controller.newMapping(CONTROLLER_MAPPING_NAME);
+mapping.from([Controller.Hardware.Keyboard.Delete]).when([Controller.Hardware.Application.PlatformWindows]).to(deleteKey);
+mapping.from([Controller.Hardware.Keyboard.Backspace]).when([Controller.Hardware.Application.PlatformMac]).to(deleteKey);
```
Comment (danteruiz) [inline scripts/system/edit.js]: Looks like people who use linux won't be able to delete entities using the short cut.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12438/HighFidelity-Beta-PR13888-f81fe6b.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12438/HighFidelity-Beta-PR13888-f81fe6b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12438/HighFidelity-Beta-Interface-PR13888-f81fe6b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12440/HighFidelity-Beta-PR13888-511478a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12440/HighFidelity-Beta-Interface-PR13888-511478a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12440/HighFidelity-Beta-PR13888-511478a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12440/HighFidelity-Beta-Interface-PR13888-511478a.exe

File: interface/src/Application.cpp [hunk 2]
```diff
@@ -909,7 +912,8 @@ bool setupEssentials(int& argc, char** argv, bool runningMarkerExisted) {
     DependencyManager::set<MessagesClient>();
     controller::StateController::setStateVariables({ { STATE_IN_HMD, STATE_CAMERA_FULL_SCREEN_MIRROR,
                     STATE_CAMERA_FIRST_PERSON, STATE_CAMERA_THIRD_PERSON, STATE_CAMERA_ENTITY, STATE_CAMERA_INDEPENDENT,
-                    STATE_SNAP_TURN, STATE_ADVANCED_MOVEMENT_CONTROLS, STATE_GROUNDED, STATE_NAV_FOCUSED } });
+                    STATE_SNAP_TURN, STATE_ADVANCED_MOVEMENT_CONTROLS, STATE_GROUNDED, STATE_NAV_FOCUSED,
+                    STATE_PLATFORM_WINDOWS, STATE_PLATFORM_MAC, STATE_PLATFORM_WINDOWS } });
```
Comment (huffman) [inline interface/src/Application.cpp]: @sam Would you be able to take a look at the controller changes and make sure they are ok?

**Comment by qaAdamSmith82** [comment]
Waiting on review confirmation then will test

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12473/HighFidelity-Beta-PR13888-96d2ba6.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12473/HighFidelity-Beta-Interface-PR13888-96d2ba6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12473/HighFidelity-Beta-PR13888-96d2ba6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12473/HighFidelity-Beta-Interface-PR13888-96d2ba6.exe

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by qaAdamSmith82** [comment]
PASS @conklin94122

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/12527/HighFidelity-Beta-PR13888-96d2ba6.apk

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/12527/HighFidelity-Beta-Interface-PR13888-96d2ba6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12527/HighFidelity-Beta-PR13888-96d2ba6.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/12527/HighFidelity-Beta-Interface-PR13888-96d2ba6.exe

**Pull request merged**